x-pack/test/functional/apps/ml/anomaly_detection/lens_to_ml.ts

```diff
@@ -0,0 +1,222 @@
+        const aggAndFieldIdentifier = 'Count(Event rate)';
+        const jobId = 'testest52';
+
+        await dashboardPreparation(selectedPanelTitle);
+
+        await testSubjects.click('embeddablePanelAction-create-ml-ad-job-action');
+
+        await retrySwitchTab(1, 10);
+        tabsCount++;
+
+        await createJobInWizard(jobId, undefined, aggAndFieldIdentifier);
+      });
+
+      it('can create multi metric job from vis with multiple compatible layers and single incompatible layer', async () => {
+        const selectedPanelTitle = 'panel3';
+        const aggAndFieldIdentifier = 'Mean(responsetime)';
+        const jobId = 'testest524';
+        const numberOfCompatibleLayers = 2;
+        const numberOfIncompatibleLayers = 1;
+
+        await dashboardPreparation(selectedPanelTitle);
+
+        await testSubjects.click('embeddablePanelAction-create-ml-ad-job-action');
+
+        await testSubjects.existOrFail('mlFlyoutLensLayerSelector');
+
+        const compatibleLayers = await testSubjects.findAll('mlLensLayerCompatible');
+        expect(compatibleLayers.length).to.eql(
+          numberOfCompatibleLayers,
+          `Expected number of compatible layers to be ${numberOfCompatibleLayers} (got '${compatibleLayers.length}')`
+        );
+
+        const incompatibleLayers = await testSubjects.findAll('mlLensLayerIncompatible');
+        expect(incompatibleLayers.length).to.eql(
+          numberOfIncompatibleLayers,
+          `Expected number of compatible layers to be ${numberOfIncompatibleLayers} (got '${incompatibleLayers.length}')`
+        );
+
+        await testSubjects.click('mlLensLayerCompatibleButton_1');
+
+        await retrySwitchTab(1, 10);
+        tabsCount++;
+
+        await createJobInWizard(jobId, undefined, aggAndFieldIdentifier);
+      });
+
+      it('shows flyout for job from vis with no compatible layers', async () => {
+        const selectedPanelTitle = 'panel4';
+        const numberOfCompatibleLayers = 0;
+        const numberOfIncompatibleLayers = 1;
+
+        await dashboardPreparation(selectedPanelTitle);
+
+        await testSubjects.click('embeddablePanelAction-create-ml-ad-job-action');
+
+        await testSubjects.existOrFail('mlFlyoutLensLayerSelector');
+
+        const compatibleLayers = await testSubjects.findAll('mlLensLayerCompatible');
+        expect(compatibleLayers.length).to.eql(
+          numberOfCompatibleLayers,
+          `Expected number of compatible layers to be ${numberOfCompatibleLayers} (got '${compatibleLayers.length}')`
+        );
+
+        const incompatibleLayers = await testSubjects.findAll('mlLensLayerIncompatible');
+        expect(incompatibleLayers.length).to.eql(
+          numberOfIncompatibleLayers,
+          `Expected number of compatible layers to be ${numberOfIncompatibleLayers} (got '${incompatibleLayers.length}')`
+        );
+      });
+
+      it('does not show link to ml with vis with no compatible layers', async () => {
```

Comment:
From the test title it's hard to spot the difference to the test before (`shows flyout for job from vis with no compatible layers`) as both say `with no compatible layers`. The actual difference is that this last test runs on an incompatible layer type while the test before runs on a compatible layer type that has been made incompatible in a different way (time shift). I'd suggest to reflect this detail in the test title, e.g. by reanming this last test title to `does not show link to ml with vis with incompatible layer type`.

Reply:
https://github.com/elastic/kibana/pull/133899/commits/b05cf28b33b439400f16dcc749d72e7edac877f0